crates/gitbutler-branch-actions/tests/dependencies.rs

```diff
@@ -731,6 +731,35 @@ fn dependecies_ignore_merge_commits() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn dependencies_handle_complex_branch_checkout() -> Result<()> {
```

Comment:
Even though I am for sure lazy with writing test descriptions, could we get a description of the test here? I've read the pr description twice, and I'm still a bit unclear on what it tests.